internal/venice-client-common/src/main/java/com/linkedin/venice/stats/metrics/MetricEntityStateThreeEnums.java

```diff
@@ -0,0 +1,195 @@
+        otelRepository,
+        registerTehutiSensorFn,
+        tehutiMetricNameEnum,
+        tehutiMetricStats,
+        baseDimensionsMap,
+        enumTypeClass1,
+        enumTypeClass2,
+        enumTypeClass3);
+  }
+
+  private void createAttributesEnumMap(
+      MetricEntity metricEntity,
+      VeniceOpenTelemetryMetricsRepository otelRepository,
+      Map<VeniceMetricsDimensions, String> baseDimensionsMap) {
+
+    Map<VeniceMetricsDimensions, String> additionalDimensionsMap = new HashMap<>();
+    for (E1 enumConst1: enumTypeClass1.getEnumConstants()) {
+      additionalDimensionsMap.put(enumConst1.getDimensionName(), enumConst1.getDimensionValue());
+      EnumMap<E2, EnumMap<E3, Attributes>> mapE2 = new EnumMap<>(enumTypeClass2);
+      attributesEnumMap.put(enumConst1, mapE2);
+      for (E2 enumConst2: enumTypeClass2.getEnumConstants()) {
+        additionalDimensionsMap.put(enumConst2.getDimensionName(), enumConst2.getDimensionValue());
+        EnumMap<E3, Attributes> mapE3 = new EnumMap<>(enumTypeClass3);
+        mapE2.put(enumConst2, mapE3);
+        for (E3 enumConst3: enumTypeClass3.getEnumConstants()) {
+          additionalDimensionsMap.put(enumConst3.getDimensionName(), enumConst3.getDimensionValue());
+          mapE3.put(
+              enumConst3,
+              otelRepository.createAttributes(metricEntity, baseDimensionsMap, additionalDimensionsMap));
+        }
+      }
+    }
+    if (attributesEnumMap.isEmpty()) {
+      throw new VeniceException(
+          "The dimensions map is empty. Please check the enum types and ensure they are properly defined.");
+    }
+  }
+
+  public Attributes getAttributes(E1 key1, E2 key2, E3 key3) {
+    if (!emitOpenTelemetryMetrics()) {
+      return null;
+    }
+    Attributes attributes = null;
+    if (key1 == null || key2 == null || key3 == null) {
+      throw new IllegalArgumentException(
+          "The key for otel dimension cannot be null for metric Entity: " + getMetricEntity().getMetricName());
+    }
+    if (!enumTypeClass1.isInstance(key1) || !enumTypeClass2.isInstance(key2) || !enumTypeClass3.isInstance(key3)) {
+      // defensive check: This only happens if the instance is declared without the explicit types
+      throw new IllegalArgumentException(
+          "The key for otel dimension is not of the correct type: " + key1.getClass() + "," + key2.getClass() + ","
+              + key3.getClass() + " for metric Entity: " + getMetricEntity().getMetricName());
+    }
+    EnumMap<E2, EnumMap<E3, Attributes>> mapE2 = attributesEnumMap.get(key1);
+    if (mapE2 != null) {
+      EnumMap<E3, Attributes> mapE3 = mapE2.get(key2);
+      if (mapE3 != null) {
+        attributes = mapE3.get(key3);
+      }
+    }
+
+    if (attributes == null) {
+      throw new IllegalArgumentException(
+          "No dimensions found for keys: " + key1 + "," + key2 + "," + key3 + " for metric Entity: "
+              + getMetricEntity().getMetricName());
+    }
```

Comment:
Seems that we don't perform this check for `MetricEntityStateOneEnum`, but here. Can we make them consistent?